drivers/test.go

```diff
@@ -319,13 +305,27 @@ func (td *TestDriver) ApplyFailure(msg *types.Message, code exitcode.ExitCode) t
 }
 
 func (td *TestDriver) applyMessageExpectCodeAndReturn(msg *types.Message, code exitcode.ExitCode, retval []byte) types.ApplyMessageResult {
-	result := td.ApplyMessage(msg)
+	result := td.applyMessage(msg)
 	if !td.validateResult(result, code, retval) {
```

Comment:
`ApplyMessage` doesn't perform any validation, it just records the values. This method should not be used by any tests so unexport it.